connection.go

```diff
@@ -31,6 +41,13 @@ type mysqlConn struct {
 	sequence         uint8
 	parseTime        bool
 	strict           bool
+	watcher          chan<- mysqlContext
+	closech          chan struct{}
+	finished         chan<- struct{}
+
+	mu          sync.Mutex // guards following fields
+	closed      error      // set non-nil when conn is closed, before closech is closed
```

Comment:
this field has type `error` but is basically just used as a `bool`/flag. Please just use a `bool` instead.